src/app/util/af-main-common.cpp

```diff
@@ -630,11 +630,11 @@ EmberStatus emAfSend(EmberOutgoingMessageType type, MessageSendDestination desti
     // tracks this.
     *messageTag        = INVALID_MESSAGE_TAG;
     EmberStatus status = EMBER_SUCCESS;
-    switch (type)
+    switch (destination.GetType())
     {
-    case EMBER_OUTGOING_VIA_BINDING: {
+    case MessageSendDestination::VariantViaBinding::VariantId: {
         EmberBindingTableEntry binding;
-        status = emberGetBinding(destination.mBindingIndex, &binding);
+        status = emberGetBinding(destination.Get<MessageSendDestination::VariantViaBinding>().mBindingIndex, &binding);
```

Comment:
We should just have GetBindingIndex(), etc APIs so consumers don't have to worry about our internal types.

Reply:
Fixed